### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by lightwalker-eth**
Proposing terminology updates that hopefully can further improve how we communicate about normalization

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,15 +12,36 @@
...
+* **beautiful name** - name that is normalizable and is equal to itself when using `ens_beautify`. This means `name == ens_beautify(name)`. For all normalizable names `ens_normalize(ens_beautify(name)) == ens_normalize(name)`.
+* **disallowed name** - name that is not normalizable. This means `ens_normalize(name)` raises a `DisallowedSequence`.
+* **curable name** - name that is normalizable, or a name in the subset of disallowed names that can still be converted into a normalized name using `ens_cure`.
+* **empty name** - name that is the empty string. An empty name is disallowed and not curable.
+* **namehashable name** - name that is ready for for use with the ENS `namehash` function. Only normalized and empty names are namehashable.
+
+**Sequences**
+* **unnormalized sequence** - a sequence from a name that is not in normalized form according to the ENS Normalization Standard.
+* **normalization suggestion** - a sequence suggested as an in-place replacement for an unnormalized sequence.
+* **normalizable sequence** - an unnormalized sequence containing a normalization suggestion that is automatically applied using `ens_normalize` and `ens_cure`.
```
Comment (Carbon225) [inline README.md:32]: Note that I changed "or" to "and".
"or" made it seem like every curable sequence is also normalizable.

**Comment by Carbon225** [comment]
* renamed all classes to match new sequence naming
* renamed class fields `disallowed_sequence_info` to `sequence_info` and `disallowed` (string) to `sequence`

File: README.md (hunk 1 above)
Comment (djstrong) [reply README.md:32]: I don't see it. `or` and `and` is ok for me. 
However, "sequence containing sequence" is a little confusing, like it is a subsequence - but maybe it my poor English.

**Comment by djstrong** [comment]
@lightwalker-eth is it ready for merge?

**Pull request merged**